app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -1,11 +1,11 @@
 class MiqServer::WorkerManagement::Kubernetes < MiqServer::WorkerManagement
-  class_attribute :current_pods
-  self.current_pods = Concurrent::Hash.new
+  attr_accessor :current_pods, :current_deployments, :deployments_monitor_thread, :pods_monitor_thread
 
-  class_attribute :current_deployments
-  self.current_deployments = Concurrent::Hash.new
-
-  attr_accessor :deployments_monitor_thread, :pods_monitor_thread
+  def initialize(*)
+    super
+    @current_pods        = Concurrent::Hash.new
+    @current_deployments = Concurrent::Hash.new
```

Comment:
Yeah I figured this would work because:

1. There are no references to these methods outside of this class
2. All of those references are from instance methods not class methods
3. The kube_monitor threads are instance vars

The one thing I'm not sure about is `EvmServer` does

```ruby
def servers_from_db
  MiqEnvironment::Command.is_podified? ? MiqServer.in_my_region.to_a : [MiqServer.my_server(true)].compact
end
```

which would return multiple `MiqServer` instances, though this would also mean multiple kube_monitor threads even if it shared a current_{pods/deployments} hash so it is likely working only on the master server or something, haven't dug into that one yet

Reply:
In a kubernetes deployment, there should only be 1 `MiqServer` in the database.

Reply:
Seems like we could simplify that method then if it will only ever return an array of one

Reply:
It's 1 MiqServer now, but that was more of a hack to fit the MiqServers / MiqRegion / Zones model...if kubernetes becomes the primary deployment type we may want to rethink it.

Reply:
https://github.com/ManageIQ/manageiq/pull/19734 looks to allow for multiple servers, maybe we're handling zones in a different way now and it is no longer needed?

Reply:
I admit I forget exactly how it's architected, but technically it's a single MiqServer, but due to the way the Server/Zone relationship works, we had to create a fake server records in order to deal with zones - there some explanation here, though it seems incomplete: https://github.com/ManageIQ/manageiq-pods/issues/353#issuecomment-566736472

Reply:
> It's 1 MiqServer now, but that was more of a hack to fit the MiqServers / MiqRegion / Zones model.

Okay, that seems like the only way the current code could work given we have class-level current_pods/current_deployments being updated by 1..N instance-level kube_monitor threads (unless I'm missing something about the "master" miq_server being the only one to run a kube_monitor)

Reply:
I have to dig back in - making Zones work was really complicated :)